### ISSUE TITLE
Create Flows

### ISSUE THREAD

**Issue by carrenelloyd** [carbon-design-system/ibm-cloud-cognitive#52]
Catalog # 5

[Link to Design](https://www.carbondesignsystem.com/community/patterns/create-flows/) (not updated yet)
[Link to interim design docs](https://ibm.ent.box.com/notes/724312709451)

Maintainers:
Andrew Smith, @andrea-gm 

Dev:
NOTE: Tearsheets are a separate issue. There are separate issues for each form of 'Create'.

## Tasks

- [ ] Initial review of design/existing code

### Working in experimental package

- [ ] Create a shell component which
  - [ ] Create ComponentName/ComponentName.js component file that renders props
        in raw form (no styling or sub components, plain text OK)
  - [ ] Adds prop-types with comments for DocGen to
        ComponentName/ComponentName.js
  - [ ] Add Doc file e.g. ComponentName.mdx with a brief description of the
        component.
- [ ] First engineering review/playback

### Continuing in experimental package

- [ ] Storybook - stories to cover design scenarios, story with long, min and
      max content, etc
- [ ] Doc tab in storybook derived from design docs with embedded stories where
      appropriate
- [ ] Code to include prop-types with comments for automated DocGen
      documentation
- [ ] Tests (written in React Test Library) to cover behavior from the
      documented in the components mdx file.

### Review and promote

- [ ] Design review
- [ ] Add design review issues addressed, issues raised or discounted.
- [ ] Engineering review - Submit PR to promote component to production package
- [ ] All engineering issues addressed, issues raised or discounted.

**Comment by vikkidpaterson** [comment]
Andrew Smith has been working on significant updates to the create pattern after collaborating with multiple teams.  My understanding it that these updates are nearly finalised and he will be providing an update to the sketch file and the documented pattern soon.
(I can't seem to assign people/labels or find Andrew to tag on this!)

**Comment by lee-chase** [comment]
It is likely @vikkidpaterson that Andrew is not a member of the carbon-design-system org. He should @ josh black in the internal slack channel.

**Comment by moores2** [comment]
**THIS QUESTION LIST IS A WORK-IN-PROGRESS**
After an initial review of the available design documentation I have the following questions:
### Major

1. The design pre-req's a `Tearsheet` component. When will the design for this be complete, and who will be implementing it?


### Minor
1. The anatomy of the modal create flow differs from a standard Carbon Modal (e.g. has a subtitle). Is this intentional (as it implies the creation of a custom type of modal for create flows)?
2. Is the side panel design slide-in or slide-over?
3. Does the side panel design have to support multi-step flows (the design implies not)?
4. Is the side panel content scrollable? If so, which part(s) e.g. all, body only, header and body...?
5. Are there any responsive design considerations for the side panel design? E.g. is it still 480px wide at mobile sizes?

**Comment by chloepoulterdesign** [comment]
Hi all, Andrew Smith has asked me to share these videos here: 

![Create_advanced_options_toggle](https://user-images.githubusercontent.com/47979346/99381237-dc511580-28c2-11eb-9e61-8b361455366b.gif)
![Create-new-topic](https://user-images.githubusercontent.com/47979346/99381244-dd824280-28c2-11eb-85fb-10c01f858208.gif)

**Comment by ampersandrew-IBM** [comment]
Thanks for adding these @chloepoulterdesign! 

@moores2, Can this motion get added as we're building out this pattern?

**Comment by moores2** [comment]
> Thanks for adding these @chloepoulterdesign!
> 
> @moores2, Can this motion get added as we're building out this pattern?

So, just to be clear - the first animation shows the left nav keeping in sync with the main content as the content is scrolled when 'show all available options' is on, and the second animation is showing the main content fading in and sliding down as the user moves from step to step, right?

**Comment by lee-chase** [comment]
Blocked by #183

**Comment by ampersandrew-IBM** [comment]
@lee-chase Can you explain why this is blocked by the Tearsheet? 

@moores2 How's the motion elements coming?

**Comment by lee-chase** [comment]
Hmm, @ampersandrew-IBM @moores2 the first planned implementation was on Tearsheets, which @moores2 was working on prior to being diverted onto other stuff.

Unblocking as the non-TearSheet implementations are not blocked.

**Comment by andrea-gm** [comment]
Created separate issues for Modal and Side panel versions and have added them to the "To do" list. Moving this general issue out of "To do" to avoid confusion.

Also I'm unclear if we need a Full page common implementation, and have not create an issue for it yet
